### PR TITLE
feat(js): improve watch mode support for tsc and swc builds

### PR DESCRIPTION
**Pull request by jaysoo**
This PR fixes watch-mode for sources and assets for `@nrwl/js:tsc` and `@nrwl/js:swc` executors.

For example, if you modified `README.md` for a buildable lib, it'll copy  it to `dist` when you update it.

## Current Behavior
 Watch mode doesn't handle assets properly

## Expected Behavior
Watch mode handles assets.

## Other Notes

* Add `--watch` option to `@nrwl/js:node` so it either executes the module once, or watch for file changes.
* Add `waitUntil`  e2e util to pick up file changes for watch-mode (See `js.test.ts`).
* Enable `serve app --no-watch` test again  in `js.test.ts`.

### CODE REVIEW

**Comment by AgentEnder** [comment]
Do we want to use chokidar or @parcel/watcher here? I know in daemon we use parcel.

**Comment by jaysoo** [comment]
> Do we want to use chokidar or @parcel/watcher here? I know in daemon we use parcel.

If it has better performance I don't see why not.

File: e2e/utils/index.ts [hunk 1]
```diff
@@ -754,3 +754,23 @@ export function expectNoAngularDevkit() {
...
+  return new Promise((resolve, reject) => {
+    const t = setInterval(() => {
+      if (predicate()) {
+        clearInterval(t);
+        resolve();
+      }
+    }, ms);
+
+    setTimeout(() => {
+      clearTimeout(t);
```
Comment (nartc) [inline e2e/utils/index.ts:772]: question: is `t` supposed to be the `setTimeout` return? or `clearTimeout()` supposed to be `clearInterval`? or this is intended?

**Comment by nartc** [comment]
Just one question

**Comment by AgentEnder** [comment]
> I don't see a way to watch a list of files though since @parcel/watcher seems to be addressing the use-case of watching any file changes in a nested directory.

Ah, that's a good point. Ignore me 😅

File: e2e/utils/index.ts (hunk 1 above)
Comment (jaysoo) [reply e2e/utils/index.ts:772]: It should be `clearInterval` to stop the interval.

**Review by nartc — APPROVED** [review]

File: packages/js/src/executors/swc/swc.impl.ts [hunk 2]
```diff
@@ -63,52 +65,71 @@ export function normalizeOptions(
...
+      join(context.root, projectRoot),
+      'package.json',
+      () =>
+        updatePackageJson(
+          options.main,
+          options.outputPath,
+          projectRoot,
+          !options.skipTypeCheck
+        )
+    );
```
Comment (AgentEnder) [inline packages/js/src/executors/swc/swc.impl.ts:103]: This watches for package JSON changes, not readme changes. Probably intended, but variable name should be refactored.

File: packages/js/src/executors/tsc/tsc.impl.ts [hunk 3]
```diff
@@ -50,47 +51,60 @@ export function normalizeOptions(
...
+  });
+
+  if (options.watch) {
+    const disposeWatchAssetChanges =
+      await assetHandler.watchAndProcessOnAssetChange();
+    const disposeReadmeChanges = await watchForSingleFileChanges(
+      join(context.root, projectRoot),
+      'package.json',
+      () => updatePackageJson(options.main, options.outputPath, projectRoot)
+    );
```
Comment (AgentEnder) [inline packages/js/src/executors/tsc/tsc.impl.ts:83]: See above comment regarding refactor

**Review by AgentEnder — CHANGES_REQUESTED** [review]
Minor nitpick

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**